### PR TITLE
Fix handling of renaming server error

### LINKED ISSUE #19: If server returns Error on renaming there are errors in console
image

### PR DESCRIPTION
**Pull request by AndriiSameliuk**
Fix #19 
image

### CODE REVIEW

**Comment by AndriiSameliuk** [comment]
@dkalinin, PTAL.

File: app/View/Component/FileItem/index.js [hunk 1]
```diff
@@ -116,6 +116,8 @@ export class FileItem extends DraggableComponent {
    * @param {String} errorMessage
    */
   set errorMessage(errorMessage) {
+    this._isRenaming = false;
```
Comment (dkalinin) [inline app/View/Component/FileItem/index.js]: When there's a server error input should be rendered with previously entered value.

**Comment by AndriiSameliuk** [comment]
@dkalinin, fixed, PTAL.

File: app/View/Component/FileItem/index.js [hunk 2]
```diff
@@ -73,21 +73,22 @@ export class FileItem extends DraggableComponent {
...
    */
   _renderInputForRenaming() {
-    const text = this._renderedFileName.textContent;
+    const text = !this._newName ? this._renderedFileName.textContent : this._newName;
     const form = document.createElement('form');
     this._div.appendChild(form);
     const input = new Input(form, 'text', 'renaming-input');
     input.value = text;
-
+    let isRenameStart = false;
```
Comment (dkalinin) [inline app/View/Component/FileItem/index.js]: isRenameStarted

Comment (dkalinin) [reply app/View/Component/FileItem/index.js]: Let's remove focusout event listener in listener for keydown.

Comment (dkalinin) [reply app/View/Component/FileItem/index.js]: Also remove input disabled setter since it's not used anymore.

**Comment by AndriiSameliuk** [comment]
@dkalinin, fixed, PTAL.

File: app/View/Component/FileItem/index.js [hunk 3]
```diff
@@ -78,23 +78,20 @@ export class FileItem extends DraggableComponent {
     this._div.appendChild(form);
     const input = new Input(form, 'text', 'renaming-input');
     input.value = text;
-    let isRenameStart = false;
+    const focusoutHandler = (() => {
```
Comment (dkalinin) [inline app/View/Component/FileItem/index.js]: Wrapping in `()` is not needed.

**Pull request merged**